warehouse/api/schema.py

```diff
@@ -0,0 +1,195 @@
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import datetime
+
+from marshmallow import Schema, fields
+
+
+class File(Schema):
+    filename = fields.Str()
+    packagetype = fields.Str()
+    python_version = fields.Str()
+    has_sig = fields.Bool(attribute="has_signature")
+    comment_text = fields.Str()
+    md5_digest = fields.Str()
+    digests = fields.Method("get_digests")
+    size = fields.Int()
+    upload_time = fields.Function(
+        lambda obj: obj.upload_time.strftime("%Y-%m-%dT%H:%M:%S")
```

Comment:
No timezone?

Reply:
Drive-by comment: You can use `fields.DateTime`. You can pass an optional format:

```python
import datetime as dt

import pytz
from marshmallow import Schema, fields

class File(Schema):
    upload_time = fields.DateTime(format="%Y-%m-%dT%H:%M:%S")


schema = File()
timezone_aware_dt = dt.datetime.utcnow()
timezone_aware_dt.replace(tzinfo=pytz.UTC)
print(schema.dump({'upload_time': timezone_aware_dt}))
# {'upload_time': '2018-06-01T13:29:25'}
```

If you don't pass a format, the default behavior is to format as ISO8601.

```python
import datetime as dt

import pytz
from marshmallow import Schema, fields

class File(Schema):
    upload_time = fields.DateTime()


schema = File()
timezone_aware_dt = dt.datetime.utcnow()
timezone_aware_dt.replace(tzinfo=pytz.UTC)
print(schema.dump({'upload_time': timezone_aware_dt}))
# {'upload_time': '2018-06-01T13:31:08.732522+00:00'}
```

Reply:
@merwok, we left out the timezone because that's what the JSON api does. However, the XML-RPC api does include a tz. I'll add tz to this field on the next push.